### ISSUE TITLE
Yandex crawler doesn't see styles in production mode

### ISSUE THREAD

**Issue by them4g** [styled-components/styled-components#2038]
I have a problem with [Yandex Webvisor](https://metrica.yandex.com/promo/webvisor/). In production mode styled-components injects styles to CSSOM. Because of this yandex bot can't render page properly (it renders page as plain html without styles).

When I disabled speedy mode in production it helps. And I'm wondering if we can make this mode to be optional (e.g. check for FORCE_DISABLE_SPEEDY enviroment variable is set to true). I'll glad if you suggest better solution.

**Comment by probablyup** [comment]
An environment variable could work. It's been discussed before to give an option to disable speedy mode... I'm not opposed. @mxstbr @kitten @imbhargav5?

**Comment by dashed** [comment]
I'm also interested in a way to control `DISABLE_SPEEDY` thru an env variable for another usecase.

**Comment by mxstbr** [comment]
What's your use case @dashed?

----

Are you sure that injecting the styles via slow mode will show up on Yandex @them4g, have you tried and verified that? 

I'm not necessarily opposed to making this a configuration option, but I'd rather keep them to a minimum.

**Comment by probablyup** [comment]
If anyone knows a dev over at Yandex they should probably also just add support for CSSOM styles since the other CSS-in-JS libraries use the same set of APIs. I know some analytics tools have upgraded to detect them.

**Comment by probablyup** [comment]
If someone at Yandex sees this, this issue requires your crawler to make use of the CSSOM APIs: https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet

In prodution, we use `insertRule` which causes the CSS to only be accessible via `document.styleSheets[0].cssRules` etc

**Comment by dashed** [comment]
> What's your use case @dashed?

@mxstbr This would sound stupid; but a segregated environment to allow styles to be editable using DevTools.

**Comment by devrelm** [comment]
@mxstbr My company has another use-case: [percy.io](https://percy.io). I'd already mentioned this [on another issue](https://github.com/styled-components/styled-components/issues/1915#issuecomment-413645595).

@probablyup's PR (#2089) is appreciated, but we'd rather not have to set up a separate build that sets that environment variable. Could an additional option be available where we could set a global variable to disable speedy? I would happily provide the PR.

Because the `DISABLE_SPEEDY` constant is set when styled-components is first loaded, we would need to set the global variable _before_ loading our app. I had imagined something like the following for my company's use-case, which could easily set up a separate subdomain pointing to the exact same staging environment, but we could also check the user-agent string or use a number of other methods.

```html
<!-- In our index.html -->
<script>window.SC_DISABLE_SPEEDY = /percy/.match(location.host)</script>
<script src="/app.min.js"></script>
```

The result would be 3 ways to disable speedy-styles:
1. Disable automatically at build time in non-prod environments by referencing the `NODE_ENV` environment variable.
2. Disable manually at build-time by setting the `SC_DISABLE_SPEEDY` environment variable.
3. Disable manually just before it initializes at run-time by setting a `SC_DISABLE_SPEEDY` global variable _before_ styled-components is initialized.

**Comment by probablyup** [comment]
Hey Mike, we used Percy extensively at zocdoc and the easy solution is just
to do your screenshots in development mode instead of production.

**Comment by devrelm** [comment]
Hi @probablyup. This doesn't necessarily work for us. We already use the `NODE_ENV` flag to hide unreleased features. If we just used Percy against a dev build, we'd end up seeing features that aren't actually released yet.

Further, we use an artifact-based release pipeline, and would prefer to have the tested build be as close as possible to the build that makes it into prod.

**Comment by devrelm** [comment]
@probablyup Yeah, this definitely doesn't work for us. We're using Create React App, which doesn't allow `NODE_ENV` to be manually set. Any builds that actually produce distributable files will _always_ have `NODE_ENV=production`.

**Comment by 0leg5ergeev** [comment]
Hi, I have the same problem on the topic on production mode on CRA, can you explain to me how you guys solved it?

**Comment by devrelm** [comment]
@0leg5ergeev 

As best as I can tell, there is no real solution for CRA apps, due to what I mentioned in my previous comment.

`DISABLE_SPEEDY` is the internal name for the constant that controls the optimization, so there wouldn't be any documentation about it. I have a PR in that provides a way to override it at runtime: #2185.

**Comment by devrelm** [comment]
@them4g as of v4.1.0, you should be able to set a `SC_DISABLE_SPEEDY` global variable _before_ loading the styled-components library to disable the speedy optimizations.

See #2185 and the v4.1.0 changelog entry for more details.

**Comment by devrelm** [comment]
@0leg5ergeev ^^
